homework/content.js

```diff
@@ -1,37 +1,50 @@
 'use strict';
 
-(function() {
-  var _wrapper = document.querySelector('#note-content-wrapper');
+(function(exports) {
 
-  function start() {
-    window.addEventListener('note-open', function(event) {
-      var note = event.detail;
-      resetWrapper();
-      drawNote(note);
-    });
-  }
+var NoteContentManager = function() {
+	// Will be an element as the content wrapper.
+	this._wrapper = null;
+};
 
-  function resetWrapper() {
-    _wrapper.innerHTML = '';
-  }
+NoteContentManager.prototype = {
+	
+	drawNote(note){
+		var title = note.title;
+		var h = document.createElement('h2');
+		h.textContent = title;
+		var passages = note.passages;
+		var buff = document.createDocumentFragment();
+		passages.forEach(function(passage) {
+			var p = document.createElement('p');
+			p.classList.add('note-passage');
+			p.textContent = passage;
+			buff.appendChild(p);
+		});
+		this._wrapper.appendChild(h);
+		this._wrapper.appendChild(buff);
+	},
+	
+	resetWrapper(){
+		this._wrapper.innerHTML = '';
+	},
+  
+	handleEvent(event) {
+		switch (event.type) {
+			case 'note-open':
+				var note = event.detail;
+				this.resetWrapper();
+				this.drawNote(note);
+				break;
+		}
+	},
 
-  function drawNote(note) {
-    var title = note.title;
-    var h = document.createElement('h2');
-    h.textContent = title;
-    var passages = note.passages;
-    var buff = document.createDocumentFragment();
-    passages.forEach(function(passage) {
-      var p = document.createElement('p');
-      p.classList.add('note-passage');
-      p.textContent = passage;
-      buff.appendChild(p);
-    });
-    _wrapper.appendChild(h);
-    _wrapper.appendChild(buff);
+  start() {
+    window.addEventListener('note-open', this);
```

Comment:
It's good to see you bind the listener with the `this`, rather than an anonymous function.